src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -2901,7 +2907,7 @@ def _create_child(self, child, parent=None):
             generic_type = type(child).__bases__[0]
             handler = self.handlers.get(generic_type)
             if not handler:
-                raise NotImplementedError()
+                raise NotImplementedError("Unsupported statement")
```

Comment:
Useful to include `type(child)` in this message?

Reply:
Yes, added.